### PR TITLE
Improve DataBuffer.h code comments

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

**Comment by majetideepak** [comment]
@xiaoxmeng can you take a look? Thanks.

**Comment by majetideepak** [comment]
@xiaoxmeng any comments?

**Comment by majetideepak** [comment]
@xiaoxmeng, can you take a look? Thanks.

File: velox/dwio/common/DataBuffer.h [hunk 1]
```diff
@@ -215,15 +216,15 @@ class DataBuffer {
...
-  T* buf_;
-  // current size
+  /// Current number of items of type T.
   uint64_t size_;
-  // maximal capacity (actual allocated memory)
+  /// Maximum capacity of items of type T.
   uint64_t capacity_;
+  /// The referenced velox buffer. 'buf_' owns the memory when 'veloxRef_' is
+  /// nullptr.
+  const velox::BufferPtr veloxRef_{nullptr};
```
Comment (xiaoxmeng) [inline velox/dwio/common/DataBuffer.h:225]: Can you keep the const members first?

File: velox/dwio/common/DataBuffer.h [hunk 2]
```diff
@@ -173,8 +173,8 @@ class DataBuffer {
   void unsafeAppend(uint64_t offset, const T* src, uint64_t items) {
     if (FOLLY_LIKELY(items > 0)) {
       std::memcpy(data() + offset, src, sizeInBytes(items));
+      size_ = (offset + items);
```
Comment (xiaoxmeng) [inline velox/dwio/common/DataBuffer.h:176]: Let's make it this change separately? This is a function change? Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
@majetideepak thanks for the improvement!

File: velox/dwio/common/DataBuffer.h (hunk 1 above)
Comment (majetideepak) [reply velox/dwio/common/DataBuffer.h:228]: done.

File: velox/dwio/common/DataBuffer.h [hunk 3]
```diff
@@ -222,14 +223,14 @@ class DataBuffer {
   }
 
   velox::memory::MemoryPool* const pool_;
-  // The referenced velox buffer. 'buf_' owns the memory when 'veloxRef_' is
-  // nullptr.
+  /// The referenced velox buffer. 'buf_' owns the memory when 'veloxRef_' is
```
Comment (xiaoxmeng) [inline velox/dwio/common/DataBuffer.h:226]: Private section just need //? Thanks!

**Review by xiaoxmeng — APPROVED** [review]

File: velox/dwio/common/DataBuffer.h (hunk 3 above)
Comment (majetideepak) [reply velox/dwio/common/DataBuffer.h]: fixed.

**Pull request closed without merging**